araproc/analysis/rpr.py

```diff
@@ -0,0 +1,206 @@
+        Standard deviation excluding the peak window.
+    """
+    bin_8 = wf_len // 8
+    pad_no_max = channel_wf.copy()
+    front_idx = max(0, int(max_bin - bin_8))
+
+    pad_no_max[front_idx:max_bin + bin_8 + 1] = np.nan
+    noise_mean = np.nanmean(pad_no_max)
+    noise_sigma = np.nanstd(pad_no_max)
+
+    if use_debug:
+        print(f"Mean: {noise_mean}, Sigma: {noise_sigma}")
+
+    return noise_mean, noise_sigma
+
+def get_ch_sliding_v2_snr_uw(channel_wf, channel_time, wf_len, use_debug=False):
+    """
+    Computes the RPR value, which is similar to SNR, for the given waveform.
+
+    Parameters
+    ----------
+    channel_wf: numpy array
+        Voltage values (squared waveform data).
+    channel_time: numpy array
+        Time values corresponding to the waveform.
+    wf_len: int
+        Total number of samples in the waveform.
+    use_debug: bool, optional
+        If True, prints debug information.
+
+    Returns
+    -------
+    rpr_val: float
+        The RPR value (signal-to-noise-like ratio) for the waveform.
+    """
+    dt =  wfu.get_dt_and_sampling_rate(channel_time)[0] 
+    channel_wf = get_sqrt_volt_sum_wf(channel_wf, dt)
+    max_bin, max_val, max_time = get_max_info(channel_wf, channel_time, use_debug)
+    noise_mean, noise_sigma = get_mean_sigma_in_no_max(channel_wf, max_bin, wf_len, use_debug)
+    rpr_val = (max_val - noise_mean) / noise_sigma
+
+    if noise_sigma <= 0:
+        print(f'Negative std detected, which may be unphysical')
+        rpr_val = -9999
+
+    if use_debug:
+        print(f"RPR: {rpr_val}, Max Val: {max_val}, Mean: {noise_mean}, Sigma: {noise_sigma}")
+
+    return rpr_val
+
+def run_rpr_calculation(waveform, use_debug=False):
+    """
+    Computes the RPR value for a given waveform by preparing the data and invoking the RPR calculation.
+
+    Parameters
+    ----------
+    Parameters
+    ----------
+    waveform: TGraph
+        A TGraph of the waveform.
+    snr: float, optional
+        Placeholder for potential SNR calculation (currently unused).
+    use_debug: bool, optional
+        If True, prints debug information.
+
+    Returns
+    -------
+    rpr_val: float
+        The calculated RPR value for the waveform.
+    """
+    channel_time, channel_wf = wfu.tgraph_to_arrays(waveform) 
+    wf_len = len(channel_wf)
+    channel_wf = channel_wf ** 2
+    channel_wf[np.isnan(channel_wf)] = 0
+    rpr_val = get_ch_sliding_v2_snr_uw(channel_wf, channel_time, wf_len, use_debug)
+    return rpr_val
+
+def get_avg_rpr(wave_bundle, chans=None, individual_antenna=False, use_debug=False):
+    """
+    Calculates channel-wise averaged RPR for a given set of waveforms.
+
+    Parameters
+    ----------
+    wave_bundle: dict of tuples
+        Dictionary containing tuples of (voltage array, time array) for each channel.
+    chans: list, optional
+        List of channels to average over. If None, averages over all channels in wave_bundle.
+    individual_antenna: bool, optional
```

Comment:
I think we can drop this parameter. If one wants the rpr for a single waveform they should just call run_rpr_calculation for that waveform.